### PR TITLE
Sync Committee memberships with Mailchimp

### PR DESCRIPTION
**Pull request by ottaviano**

### CODE REVIEW

File: app/config/services/messenger.yml [hunk 1]
```diff
@@ -76,3 +76,9 @@ services:
     messenger.receiver_locator:
         tags: [container.service_locator]
         arguments: [[]]
+
```
Comment (JeremieSamson) [inline app/config/services/messenger.yml]: can you remove this line please :) ?

File: src/Mailchimp/Driver.php [hunk 2]
```diff
@@ -88,6 +88,31 @@ public function sendCampaign(string $externalId): bool
         return $this->sendRequest('POST', sprintf('/campaigns/%s/actions/send', $externalId));
     }
 
+    public function createStaticSegment(string $name): array
+    {
+        $response = $this->send('POST', sprintf('/lists/%s/segments', $this->listId), [
```
Comment (JeremieSamson) [inline src/Mailchimp/Driver.php]: Shouldn't we have same indentation for those 3 functions ?

File: templates/profiler/messenger.html.twig [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                <span>{{ collector.exceptionsCount }}</span>
+            </span>
+        {% endif %}
+    </span>
+{% endblock %}
+
+{% block head %}
+    {{ parent() }}
+    <style>
+        .message-item thead th { position: relative; cursor: pointer; user-select: none; padding-right: 35px; }
```
Comment (thomasbisignani) [inline templates/profiler/messenger.html.twig]: Is it possible to move this style in a SCSS file ?

File: app/config/services/messenger.yml (hunk 1 above)
Comment (ottaviano) [reply app/config/services/messenger.yml]: done, thanks 👍

File: templates/profiler/messenger.html.twig (hunk 3 above)
Comment (ottaviano) [reply templates/profiler/messenger.html.twig]: I prefer do not change this file, as it comes from the package Symfony/FrameworkBundle:4.1 and will be removed when we will up to sf4

File: src/Mailchimp/Synchronisation/EventListener/AdherentEventSubscriber.php [hunk 4]
```diff
@@ -64,14 +66,31 @@ public function onAfterUpdate(UserEvent $event): void
...
+        $this->dispatchMessage($event->getUser()->getUuid(), $event->getUser()->getEmailAddress());
+    }
+
+    public function onCommitteePrivilegeChange(CommitteeEventInterface $event): void
+    {
+        $adherent = $event->getAdherent();
+
+        $this->dispatchMessage($adherent->getUuid(), $adherent->getEmailAddress());
+
+        if ($committee = $event->getCommittee()) {
```
Comment (Remg) [inline src/Mailchimp/Synchronisation/EventListener/AdherentEventSubscriber.php]: what about:
```
if (!$committee = $event->getCommittee()) {
    return;
}
```

to remove an indentation level?
(not important)

File: tests/Admin/CommitteeAdminTest.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+{
+    use ControllerTestTrait;
+    use MessengerTestTrait;
+
+    /**
+     * @var CommitteeRepository
+     */
+    private $committeeRepository;
+
+    public function test committee static segment command is dispatched when committee is approved(): void
```
Comment (Remg) [inline tests/Admin/CommitteeAdminTest.php]: `public function dude_what?`

**Pull request merged**